src/react/DeviceStatusPage/index.tsx

```diff
@@ -0,0 +1,471 @@
+  isAddingAlias: boolean;
+};
+
+type EditAliasModalT = {
+  kind: 'edit_alias_modal';
+  isUpdatingAlias: boolean;
+  targetAlias: string;
+};
+
+type DeleteAliasModalT = {
+  kind: 'delete_alias_modal';
+  isDeletingAlias: boolean;
+  aliasKey: string;
+  aliasValue: string;
+};
+
+type NewMetadataModalT = {
+  kind: 'new_metadata_modal';
+  isAddingMetadata: boolean;
+};
+
+type EditMetadataModalT = {
+  kind: 'edit_metadata_modal';
+  isUpdatingMetadata: boolean;
+  targetMetadata: string;
+};
+
+type DeleteMetadataModalT = {
+  kind: 'delete_metadata_modal';
+  isDeletingMetadata: boolean;
+  metadataKey: string;
+  metadataValue: string;
+};
+
+function isWipeCredentialsModal(modal: PageModal): modal is WipeCredentialsModalT {
+  return modal.kind === 'wipe_credentials_modal';
+}
+
+function isAddToGroupModal(modal: PageModal): modal is AddToGroupModalT {
+  return modal.kind === 'add_to_group_modal';
+}
+
+function isNewAliasModal(modal: PageModal): modal is NewAliasModalT {
+  return modal.kind === 'new_alias_modal';
+}
+
+function isEditAliasModal(modal: PageModal): modal is EditAliasModalT {
+  return modal.kind === 'edit_alias_modal';
+}
+
+function isDeleteAliasModal(modal: PageModal): modal is DeleteAliasModalT {
+  return modal.kind === 'delete_alias_modal';
+}
+
+function isNewMetadataModal(modal: PageModal): modal is NewMetadataModalT {
+  return modal.kind === 'new_metadata_modal';
+}
+
+function isEditMetadataModal(modal: PageModal): modal is EditMetadataModalT {
+  return modal.kind === 'edit_metadata_modal';
+}
+
+function isDeleteMetadataModal(modal: PageModal): modal is DeleteMetadataModalT {
+  return modal.kind === 'delete_metadata_modal';
+}
+
+type PageModal =
+  | WipeCredentialsModalT
+  | AddToGroupModalT
+  | NewAliasModalT
+  | EditAliasModalT
+  | DeleteAliasModalT
+  | NewMetadataModalT
+  | EditMetadataModalT
+  | DeleteMetadataModalT;
+
+interface Props {
+  astarte: AstarteClient;
+  deviceId: string;
+}
+
+export default ({ astarte, deviceId }: Props): React.ReactElement => {
+  const deviceFetcher = useFetch(() => astarte.getDeviceInfo(deviceId));
+  const groupsFetcher = useFetch(() => astarte.getGroupList());
+  const [activeModal, setActiveModal] = useState<PageModal | null>(null);
+
+  const unjoinedGroups = useMemo(() => {
+    if (deviceFetcher.status === 'ok' && groupsFetcher.status === 'ok') {
+      const joinedGroups = new Set(deviceFetcher.value.groups);
+      return groupsFetcher.value.filter((groupName) => !joinedGroups.has(groupName));
+    }
+    return [];
+  }, [deviceFetcher.status, groupsFetcher.status]);
+
+  const dismissModal = useCallback(() => {
+    setActiveModal(null);
+  }, []);
+
+  const inhibitDeviceCredentialsRequests = useCallback(
+    (inhibit) => {
+      astarte
+        .inhibitDeviceCredentialsRequests(deviceId, inhibit)
+        .then(() => {
+          deviceFetcher.refresh();
+        })
+        .catch(() => {
+          // TODO add alert
```

Comment:
Is this a TODO for later or you intended to add an implementation before merging the PR? Same for other TODOs below

Reply:
good catch! i totally forgot about it